### PR TITLE
[APPC-4359] Implementation Paypal Custom Tabs 

### PR DESCRIPTION
**Pull request by gustavoaptoide**
**What does this PR do?**

 Change PayPal to webView redirect in custom tabs for user to save user credential, improving user experience

**Database changed?**

    No

**How should this be manually tested?**

  Try to make a purchase in Paypal with V2, V1 and inside TopUp, Iap flow

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-4359](https://aptoide.atlassian.net/browse/APPC-4359)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

[APPC-4359]: https://aptoide.atlassian.net/browse/APPC-4359?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ricardo-frango — CHANGES_REQUESTED** [review]
I tried to buy with paypal v2 and it's not leaving the custom tab after logging in. when we leave the custom tab, it stays in infinite loading

File: app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentViewModel.kt [hunk 1]
```diff
@@ -99,6 +102,8 @@ class AdyenPaymentViewModel @Inject constructor(
   val singleEventState = _singleEventState.receiveAsFlow()
   lateinit var paymentStateEnum: PaymentStateEnum
   var cancelPaypalLaunch = false
+  var runningCustomTab = false
```
Comment (ricardo-frango) [inline app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentViewModel.kt:105]: this is always false

File: app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentViewModel.kt [hunk 2]
```diff
@@ -954,6 +959,27 @@ class AdyenPaymentViewModel @Inject constructor(
     )
   }
 
+  fun processPayPalResult(paymentData: Observable<AdyenComponentResponseModel>) {
+    if (isFirstResultRun) {
```
Comment (ricardo-frango) [inline app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentViewModel.kt:963]: this seems a strange behavior. maybe the problem lies on another place for adding this logic...

File: app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalIABViewModel.kt [hunk 3]
```diff
@@ -170,6 +181,46 @@ class PayPalIABViewModel @Inject constructor(
...
+  fun openUrlCustomTab(context: Context, url: String) {
+    if (runningCustomTab) return
+    runningCustomTab = true
+    val customTabsBuilder = CustomTabsIntent.Builder().build()
+    customTabsBuilder.intent.setPackage(GooglePayWebFragment.CHROME_PACKAGE_NAME)
+    customTabsBuilder.launchUrl(context, Uri.parse(url))
+  }
+
+  fun processPayPalResult(amount: BigDecimal, currency: String, transactionBuilder: TransactionBuilder, origin: String?) {
+    if (isFirstResultRun) {
```
Comment (ricardo-frango) [inline app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalIABViewModel.kt:193]: this seems a strange behavior. maybe the problem lies on another place for adding this logic...

File: app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalTopupViewModel.kt [hunk 4]
```diff
@@ -70,6 +86,37 @@ class PayPalTopupViewModel @Inject constructor(
     )
   }
 
+  fun processPayPalResult(amount: String, currency: String) {
+    if (isFirstResultRun) {
```
Comment (ricardo-frango) [inline app/src/main/java/com/asfoundation/wallet/billing/paypal/PayPalTopupViewModel.kt:90]: this seems a strange behavior. maybe the problem lies on another place for adding this logic...

File: app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentViewModel.kt (hunk 1 above)
Comment (gustavoaptoide) [reply app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentViewModel.kt:105]: This var is changed inside Fragment When lauch CustomTabs

**Pull request closed without merging**